### DISCUSSION TITLE
[Important] Repository status

### DISCUSSION THREAD

**Discussion by EdricChan03** [EdricChan03/studybuddy-android#373] [Repo Announcements]
## TL;DR:

- Less frequent maintenance
- Cutting down on project scope
  - Resolve technical debt
    - Refactor the app before re-expanding the scope
- Next release is not coming anytime soon

---

## Maintenance

As I'm currently taking tertiary education, I'm unable to actively maintain this repository until further notice.

I'll still try my best to maintain it semi-regularly but expect less activity from me moving forward.

## Project scope

I'm also planning on cutting down on the scope of the project. Originally I had planned to:
- Rework the tasks feature
- Implement a calendar
- Implement a chat feature
- Implement a quiz feature
- Implement 3rd party integrations (e.g. Google Tasks support)
- Rework the app to use Android Jetpack libraries (e.g. ViewModel)
- Refactor the migrated Java code (e.g. the task fragment is still pretty messy)

However, the amount of effort required to implement all the features above in the amount of free time I have would not be worth it. This also leads to technical debt from the weird coding practices I had used back in the early days of this project from having to migrate over to use proper design patterns (for example the Model-View-ViewModel and repository patterns).

As such, I will cut down on the scope of the project and only currently focus on reworking the app to use better practices (i.e. refactor to use Android Jetpack libraries and improve migrated Java code).

I'm currently working on migrating the app to use ViewModel as well as Kotlin coroutines! Stay tuned :)

## Releases

A next release would not be coming any time soon while I refactor the code. In the meantime, you can either build the application locally or get a nightly build from the "Nightly releases" GitHub Actions workflows. However, note that bugs are to be expected so report them here if you can!
